### PR TITLE
Add archive page for past events. Fixes #304.

### LINKED ISSUE #304: Archive Page For Past Events
**As**
pupil

**I want**
an archive page for past events

**In order to**
be able to browse past events

**Dependencies:**

**Estimate:**

**Acceptance Criteria:**
- [ ] archive page should have the same design as the event overview page (except apply buttons)
- [ ] should be reachable via button on the overview page ("Vergangene Veranstaltungen")
- [ ] button on the left over first event (in the same "line" as the new button for organisers)
- [x] overview page should now be sorted like main page so that the next events are displayed at the top (already done in #316)

### PR DESCRIPTION
**Pull request by tom95**

### CODE REVIEW

File: spec/factories/date_ranges.rb [hunk 1]
```diff
@@ -28,5 +28,11 @@
             start_date Date.tomorrow
             end_date Date.tomorrow
         end
+
+        trait :force_past do
```
Comment (bjrne) [inline spec/factories/date_ranges.rb]: I'm not completely sure about RoR naming conditions, but I think "Entity :withX :withoutY" etc reads better than these titles.

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -6,14 +6,18 @@
 
 <div class="row">
   <div class="event-list col-md-10 col-md-offset-1">
-    <% if can? :new, Event %>
     <div class="clearfix">
       <%= link_to new_event_path, :class => 'btn btn-primary pull-right' do
         glyph('plus') + ' ' +
         t('.new', :default => t('helpers.links.new'))
+      end if can? :new, Event %>
```
Comment (MasterCarl) [inline app/views/events/index.html.erb:13]: Please add a newline between `end` and `if`

File: app/views/events/archive.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<div class="row">
+  <div class="event-list col-md-10 col-md-offset-1">
+    <%= link_to events_path, :class => 'btn btn-default' do
+      glyph('chevron-left') + ' ' +
+      t('.new', :default => t('events.all_events'))
```
Comment (MasterCarl) [inline app/views/events/archive.html.erb]: Why is the translation key `.new`?

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -254,6 +256,13 @@ def event_params
 
     end
 
+    def add_event_query_conditions(query)
```
Comment (MasterCarl) [inline app/controllers/events_controller.rb:259]: I'm not too happy with the naming here, but then again, I don't have any better ideas, either.

**Review by MasterCarl — CHANGES_REQUESTED** [review]
Overall, this looks good

File: app/views/events/index.html.erb (hunk 2 above)
Comment (tom95) [reply app/views/events/index.html.erb:13]: this is actually a "suffix" if. believe me if I tell you that it was even more confusing the other way around

File: app/controllers/events_controller.rb (hunk 4 above)
Comment (corinnaj) [reply app/controllers/events_controller.rb:259]: I think the naming is fine.

File: app/views/events/index.html.erb (hunk 2 above)
Comment (MasterCarl) [reply app/views/events/index.html.erb:13]: Oh, alright

**Review by MasterCarl — APPROVED** [review]

**Pull request merged**